From 0a37e67f56d33e96c73d2671251b5c2f11efb394 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 12 Sep 2024 16:00:31 +0200
Subject: [PATCH 0162/1010] net: dovetail: deny expanding oob buffer heads

Since oob buffer heads may be handled from the oob stage, we cannot
dynamically expand their storage safely. Make sure pskb_expand_head()
got the memo, denying requests from oob callers.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 net/core/skbuff.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 29b1803e4f..325c66ec00 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -1967,6 +1967,10 @@ int pskb_expand_head(struct sk_buff *skb, int nhead, int ntail,
 
 	BUG_ON(skb_shared(skb));
 
+	/* skb_reserve() is your friend for out-of-band operations. */
+	if (skb_is_oob(skb))
+		return -ENOMEM;
+
 	skb_zcopy_downgrade_managed(skb);
 
 	if (skb_pfmemalloc(skb))
-- 
2.30.2

